code/math/vecmat.cpp

```diff
@@ -1015,7 +1015,13 @@ angles *vm_extract_angles_matrix(angles *a, const matrix *m)
 	else											//cosine is larger, so use it
 		cosp = m->vec.fvec.xyz.z*cosh;
 
-	a->p = atan2_safe(-m->vec.fvec.xyz.y, cosp);
+	//using the fvec_xz_distance extracts the correct pitch from the matrix --wookieejedi
+	//previously cosp was used as the denominator, but this resulted in some incorrect pitch extractions
+	float fvec_xz_distance;
+
+	fvec_xz_distance = pow((pow((m->vec.fvec.xyz.x), 2.0f) + pow((m->vec.fvec.xyz.z), 2.0f)), 0.5f);
```

Comment:
Try to avoid the `pow` function if at all possible, since it is slow.  This is especially important in the vector math code which is used everywhere, all the time.

First, replace pow(n, 2.0) with n*n.  Then see if the formula can be refactored to use the distance-squared.  If not, use fl_sqrt because that is more efficient than pow(n, 0.5).

Reply:
Thanks, unfortunately with everything I tried I was unable to refactor to remove the sqrt.